Fix broken Storybook: Change ng-template to ng-container for ngIf and rename icon directive

jlf0dev opened the pull request:
## Type of change

<!-- (mark with an `X`) -->

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

<!--Describe what the purpose of this PR is. For example: what bug you're fixing or what new feature you're adding-->

While working on #4634 I found that our simple dialog throws an angular error saying that `*ngIf's cannot be used on an ng-content tag`. To fix this, I added an `ng-template` around the `ng-content` with the `*ngIf`. Unfortunately, that broke the icon on the dialog and it wasn't showing anymore.

~This PR reverts that change and puts the `*ngIf` back on the `ng-content`.~ I also went ahead and renamed our icon directive to follow other directives naming convention, camel-case.

~For now I think we can deal with the error until we find a better way to handle that situation.~

This will also fix our storybook issues.

Edit:
I was using a `ng-template` instead of a `ng-container`, switching that fixed the issue with the icon.

I also fixed a regression in the product switcher and fixed the org vault showing the incorrect text color on the bulk delete button.

## Code changes

<!--Explain the changes you've made to each file or major component. This should help the reviewer understand your changes-->
<!--Also refer to any related changes or PRs in other repositories-->

- **file.ext:** Description of what was changed and why

## Screenshots

<!--Required for any UI changes. Delete if not applicable-->

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

Hinton reviewed (approved):
It should probably have been ng-container. template is used for content projection.

There are some other UI changes related to focus @willmartian any idea why Chromatic had them focused previously?

willmartian commented:
> There are some other UI changes related to focus @willmartian any idea why Chromatic had them focused previously?

The product switcher is no longer receiving focus on open (it should); regression from https://github.com/bitwarden/clients/pull/4634 with `cdkTrapFocusAutoCapture` being removed. @jlf0dev is taking a look at it (thanks!) 👍

jlf0dev commented:
Made some changes to this PR:
- Per @Hinton, switched out the `ng-template` for an `ng-container`, fixing the icon issue I caused
- Per @willmartian, added the `ariaRole` check to the menu component, solving the regression to the product switcher
